File: practice/convert/example2_book_library/java/book-library/src/main/java/dev/starryeye/book_library/controller/user/response/UsersResponse.java
package dev.starryeye.book_library.controller.user.response;

import dev.starryeye.book_library.application.user.query.result.GetUsersResult;

import java.util.List;

public record UsersResponse(
        List<UserResponse> users
) {

    public static UsersResponse of(GetUsersResult result) {
        return new UsersResponse(
                result.users().stream()
                        .map(userResult -> new UserResponse(
                                userResult.id(),
                                userResult.username(),
                                userResult.age()
                        ))
                        .toList()
        );
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/user/response/UsersResponse.kt
package dev.starryeye.book_library.controller.user.response

import dev.starryeye.book_library.application.user.query.result.GetUsersResult

data class UsersResponse(
    val users: List<UserResponse>,
) {

    companion object {
        fun of(usersResult: GetUsersResult): UsersResponse {
            return UsersResponse(
                usersResult.users.map { userResult -> UserResponse(
                    userResult.id,
                    userResult.username,
                    userResult.age
                ) }
            )
        }
    }
}